CacheMultimap.adoc

```diff
@@ -0,0 +1,79 @@
+= Native cache Mutimap
+
+The goal is to provide Infinispan Native CacheMultimap support. The proposal here is to start with the smallest interface implementation, in order to provide a proper API to the first multimap client, vert-x project.
+However, this API will evolve, and this first implementation must work on Embedded and Server mode.
+This first Multimap won't support duplicate values on the same key. 
+
+[source, java]
+.CacheMultimap.java
+----
+public interface CacheMultimap<K, V> {
+   boolean put(K key, V value); // <1> <2>
+
+   Collection<V> get(K key); // <3>
+
+   boolean remove(K key); //<4>
```

Comment:
@karesti And what are you trying to solve? "void put" or "void remove" just don't force the code to load the collection into memory (though in practice you'll have to, as you can't send a function to the cache store), and not force to transfer any data over the wire.
So you can expose:
1. void put(), void remove()
2. boolean putAndReturnTrueIfSomeDataWasPresent(), boolean removeAndReturnTrueIfSomeEntryWasPresent()
3. the same as 2 but returning integers with collection sizes
4. Collection<V> putAndReturnAllPreviousEntries(), Collection<V> removeAndReturnThoseRemovedEntries();
5. combination of 1. and 2./3. and 4.

Since you have to load the data anyway and the cost of marshalling booleans/integers is neglible, I personally like 3. the most (4. could eventually get in *in addition to 3.* but not necessarily in the first draft). But I don't mind any variant if there's other option than 4.